server/routerlicious/packages/services-core/src/runWithRetry.ts

```diff
@@ -0,0 +1,41 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { delay } from "@fluidframework/common-utils";
+import { ILogger } from "./lambdas";
+
+export async function runWithRetry<T>(
+    api: () => Promise<T>,
+    callName: string,
+    maxRetries: number,
+    retryAfterMs: number,
+    logger?: ILogger,
+    shouldRetry?: (error) => boolean,
+): Promise<T | undefined> {
+    let result: T | undefined;
+    let retryCount = 0;
+    let success = false;
+    do  {
+        try {
+            result = await api();
+            success = true;
+        } catch (error) {
+            logger?.info(`Error running ${callName}: retryCount ${retryCount}, error ${error}`);
+            if (shouldRetry !== undefined && shouldRetry(error) === false) {
+                logger?.info(`Should not retry ${callName}`);
+                break;
```

Comment:
should this be return Promise.reject(error)?

Reply:
In this case since we want to ignore the error, we do not want to reject - added a shouldIgnoreError method param as discussed offline, to make this clearer :)